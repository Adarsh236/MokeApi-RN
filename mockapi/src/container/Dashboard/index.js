import React from 'react';
import {Button, View, Text, StyleSheet} from 'react-native';

console.log('Hi from React Native');

const Dashboard = ({navigation}) => {
  return (
    <View style={styles.dashboardArea}>
      <Text style={{fontSize: 25}}>Star war Character list</Text>
      <Button
        onPress={() => navigation.navigate('PersonList')}
        title="GET STARTED"
        color="#006400"
        style={styles.startButton}
      />
    </View>
  );
};

export default Dashboard;

const styles = StyleSheet.create({
  dashboardArea: {
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
