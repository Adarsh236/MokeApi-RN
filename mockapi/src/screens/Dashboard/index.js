import React from 'react';
import {Button, View, Text, StyleSheet, SafeAreaView} from 'react-native';
import {NetworkConsumer} from 'react-native-offline';
import {
  checkInternetConnection,
  offlineActionCreators,
} from 'react-native-offline';

console.log('Dashbroad');

function Separator() {
  return <View style={styles.separator} />;
}

const Dashboard = ({navigation}) => {
  return (
    <SafeAreaView style={styles.container}>
      <View>
        <Text style={styles.title}>Star war Character list</Text>

        <Separator />
        <Button
          onPress={() => navigation.navigate('PersonListTimer')}
          title="Person List With Timer"
          color="#006400"
        />
        <Separator />
        <Button
          onPress={() => navigation.navigate('PersonListNetInfo')}
          title="Person List With NetInfo"
          color="#006400"
        />
        <Separator />

        <NetworkConsumer>
          {({isConnected}) =>
            isConnected ? (
              <Text style={styles.text}>You are online</Text>
            ) : (
              <Text style={styles.text}>You are offline</Text>
            )
          }
        </NetworkConsumer>
      </View>
    </SafeAreaView>
  );
};

export default Dashboard;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 10,
    marginHorizontal: 16,
    backgroundColor: '#000000',
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    textAlign: 'center',
    marginVertical: 8,
    fontSize: 25,
    color: '#FFFFFF',
  },
  text: {
    textAlign: 'center',
    marginVertical: 8,
    fontSize: 20,
    color: '#FFFFFF',
  },
  fixToText: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  separator: {
    marginVertical: 8,
    borderBottomColor: '#737373',
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
});
