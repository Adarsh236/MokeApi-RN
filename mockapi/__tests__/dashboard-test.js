// npm test
// npm test -- -u

import 'react-native';
import React from 'react';
import Dashboard from '../src/container/Dashboard';
import renderer from 'react-test-renderer';

test('Dashboard snapShot', () => {
  const snap = renderer.create(<Dashboard />).toJSON();
  expect(snap).toMatchSnapshot();
});
