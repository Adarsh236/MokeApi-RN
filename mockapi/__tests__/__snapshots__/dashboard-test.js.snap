// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 25,
      }
    }
  >
    Star war Character list
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#006400",
            },
          ]
        }
      >
        GET STARTED
      </Text>
    </View>
  </View>
</View>
`;
